### PR TITLE
Add Mistral, use "ai" lib

### PR DESCRIPTION
**Pull request by laulauland**
This PR consolidates interfaces with AI providers using the `ai` library from Vercel and adds support for Mistral.

### CODE REVIEW

File: api/src/routes/settings.ts [hunk 1]
```diff
@@ -31,10 +31,10 @@ app.post("/v0/settings", cors(), async (ctx) => {
 
   const parsedContent = SettingsSchema.parse(content);
   // Remove the stored api key if the feature is disabled
-  if (!parsedContent.aiEnabled) {
-    parsedContent.openaiApiKey = undefined;
-    parsedContent.anthropicApiKey = undefined;
-  }
+  // if (!parsedContent.aiEnabled) {
```
Comment (brettimus) [inline api/src/routes/settings.ts:34]: i'd say either remove this chunk or uncomment

part of the logic behind this was to allow the end user to delete api keys... but i don't think that worked anyhow

File: studio/src/pages/SettingsPage/AISettingsForm.tsx [hunk 2]
```diff
@@ -220,16 +227,18 @@ export function AISettingsForm({
...
                                       ]
                                     }{" "}
                                     API. It should be ignored by version control
                                     by default.
                                   </FormDescription>
                                   <FormControl>
                                     <ApiKeyInput
-                                      value={field.value ?? ""}
-                                      onChange={field.onChange}
+                                      value={(field.value as string) ?? ""}
```
Comment (brettimus) [inline studio/src/pages/SettingsPage/AISettingsForm.tsx:238]: oh weird was this broken?

File: studio/src/pages/SettingsPage/AISettingsForm.tsx [hunk 3]
```diff
@@ -220,16 +227,18 @@ export function AISettingsForm({
...
+                                      onChange={(
```
Comment (brettimus) [inline studio/src/pages/SettingsPage/AISettingsForm.tsx]: was this handler broken before? 😓

Comment (laulauland) [reply studio/src/pages/SettingsPage/AISettingsForm.tsx]: I realize I need to revisit this in light of the settings model changes

**Review by brettimus — APPROVED** [review]
approved just needs to fix the things i identified when testing 

we will probably wanna check out ade's pr after this too, to look for possible regressions ¯\_(ツ)_/¯

**Comment by brettimus** [comment]
two things: 

1. Don't forget to cut a new release of the types package
2. Changelog!

File: api/src/lib/ai/tools.ts [hunk 4]
```diff
@@ -1,76 +1,46 @@
+import { z } from "zod";
+
```
Comment (brettimus) [inline api/src/lib/ai/tools.ts:2]:
```suggestion

// NOTE - We cannot use `.optional` from zod because it does not play nicely with structured output
```

**Pull request merged**